index.js

```diff
@@ -185,9 +193,53 @@ class Ora {
 			cliCursor.show(this.stream);
 		}
 
+		if (this.consumeStdin && process.stdin.isTTY) {
+			this.stopConsumingStdin();
+		}
+
 		return this;
 	}
 
+	startConsumingStdin() {
+		const me = this;
+		const {stdin} = process;
+
+		this._stdinOldRawMode = stdin.isRaw;
+		this._stdinOldEmit = stdin.emit;
+		this._stdinOldEmitOwnProperty = Object.prototype.hasOwnProperty.call(stdin, 'emit');
+
+		stdin.setRawMode(true);
+		stdin.on('data', noop);
+		stdin.emit = function (event, data) {
+			if (event === 'data') {
+				if (data.includes(0x03)) {
+					if (this.listenerCount('SIGINT') > 0) {
+						this.emit('SIGINT');
+					} else {
+						process.emit('SIGINT');
+					}
+				}
+			} else {
+				me._stdinOldEmit.apply(this, arguments); // eslint-disable-line prefer-rest-params
+			}
+		};
+	}
+
+	stopConsumingStdin() {
+		const {stdin} = process;
+		stdin.setRawMode(this._stdinOldRawMode);
+		stdin.off('data', noop);
+		if (this._stdinOldEmitOwnProperty) {
+			stdin.emit = this._stdinOldEmit;
+		} else {
+			delete stdin.emit;
```

Comment:
I don't understand the logic here. If the user calls `stopConsumingStdin()` with calling `startConsumingStdin()`, it will delete the process emit method.

Reply:
It is internal function, so user shouldn't ever be calling this by himself.

As for what it does when called normally(by `ora`), it restores stdin object to the state it was in before we overridden `emit`, which is without `emit` property normally, but there is a check if it was own property above.

Reply:
Right, I meant `stop()`, not `stopConsumingStdin(). So if the user calls `stop()` before `start()` is called it will delete the method which is not ours. We should prevent mistakes like that.

> which is without emit property normally

```
❯ node
Welcome to Node.js v12.3.1.
Type ".help" for more information.
> process.stdin.emit
[Function]
```